### DISCUSSION TITLE
Consider adding a Spacer (or "separator") to Options, cursor should be jumping over it like there's no tomorrow

### DISCUSSION THREAD

**Discussion by rusq** [charmbracelet/huh#595] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Would be great to have the ability to add a spacer to the Options, i.e.:

<img width="338" alt="image" src="https://github.com/user-attachments/assets/362f4c27-8df0-4aad-a44e-7bff61c2e504">

**Describe the solution you'd like**
I'd like to be able to add a menu (options) separator, which wouldn't be highlighted by cursor.  Cursor should jump over it, pretty much the same behaviour as the usual drop-down OS menus, i.e. 

<img width="384" alt="image" src="https://github.com/user-attachments/assets/d10bba96-b696-4e62-9263-9df7b61f5e5d">

**Describe alternatives you've considered**
Right now, I'm just adding a `strings.Repeat`, with an empty option value, and when processing, I ignore it, but that presents an inconvenience to a "hypothetical" user.  In the first image, it would take only two keypresses to reach the "Exit" Option from "Run" Option, but currently, it takes three.

The other alternative is not to include spacer, which hinders the UX, in my opinion.

**Additional context**
Maybe I should be writing custom bubbles for my needs, but huh is just so convenient!

And maybe I'm missing something.

Thank you for consideration.

**Comment by bashbunni** [comment, marked as answer]
Hey @rusq if you want full customization like this, you might be better off writing out a little Bubble Tea program. That said, with the example you've shown above, I would personally just split out the "Exit to Main Menu" as a Confirm field or something. Basically just grouping related behaviour by field instead of trying to divide up content within a field.

As we're tidying up the backlog of this project, I'm going to convert this to a discussion. Feel free to respond with your thoughts or questions :)

**Comment by rusq** [comment]
Hey @bashbunni , thanks for the response, that's exactly how I did this - ended up writing a small bubble for that, and use instead of huh in "complex" cases, i.e. 
<img width="558" alt="image" src="https://github.com/user-attachments/assets/21b91c7d-cdcd-416c-b720-9cbf777a6e22" />

I think this can be closed anyway.  Thanks for the great library :)

**Comment by rusq** [comment]
For the benefit of other readers (GPL v3 license):
- menu [bubble](https://github.com/rusq/slackdump/tree/2e6a7f3375836d6f4ab1060e2a5247cbab673bac/cmd/slackdump/internal/ui/bubbles/menu)
- [sample from the screenshot](https://github.com/rusq/slackdump/blob/2e6a7f3375836d6f4ab1060e2a5247cbab673bac/cmd/slackdump/internal/workspace/workspaceui/workspaceui.go#L75)
